pygfx/geometries/_box.py

```diff
@@ -52,12 +52,16 @@ def box_geometry(
     plane_csys = np.array(
         [
             *cube_normal_up[4],
-            np.cross(*cube_normal_up[4]),
+            np.cross(
+                np.ndarray.astype(cube_normal_up[4, 0], np.float32),
+                np.ndarray.astype(cube_normal_up[4, 1], np.float32)
```

Comment:
Can you pass `copy=False` as well? That way no copy will be made if the input is already float32

Reply:
Yes - will do when I get back home

Reply:
Done